[JIT] add torch.jit.is_scripting() api

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25263 [JIT] add torch.jit.is_scripting() api**
* #25262 [JIT] preserve ignored function return value type

This adds an api to return true in script and false in eager, which together with @ignore allows guarding of not yet supported JIT features. Bikeshedding requested please. 

cc @zou3519 

```
def foo():
   if not torch.jit.is_scripting():
      return torch.linear(...)
   else:
      return addmm(...)
```

Differential Revision: [D17273136](https://our.internmc.facebook.com/intern/diff/D17273136)

Commit:
add torch.jit.is_scripting() api

suo commented on torch/jit/__init__.py:
```diff
@@ -1741,6 +1741,10 @@ def _unwrap_optional(x):
     return x
 
+# primitive that returns True when in compilation and False otherwise
+def _is_scripting():
```
If you intend to use this in library code like torchvision, it should be considered part of our public interface (and thus not prefixed by `_`).

Commit:
Update on "[JIT] add torch.jit.is_scripting() api"


This adds an api to return true in script and false in eager, which together with @ignore allows guarding of not yet supported JIT features. Bikeshedding requested please. 

cc @zou3519 

```
def foo():
   if not torch.jit.is_scripting():
      return torch.linear(...)
   else:
      return addmm(...)
```

eellison commented:
It was actually https://github.com/pytorch/pytorch/pull/25262 that landed - i think ghimport got messed up because i previously landed that PR and it got reverted. I will submit this as a separate PR and try to land it today.
